src/ocean/core/BitArray.d

```diff
@@ -189,13 +191,8 @@ struct BitArray
     }
     body
     {
-        size_t mDim=len/32;
-        ptr[0..mDim] = rhs.ptr[0..mDim];
-        int rest=cast(int)(len & cast(size_t)31u);
```

Comment:
> Since I'm not familiar with BitArray I can't figure out from top of my head what this was doing, but it wasn't bit by bit copying as your commit message suggests.

True, the commit message needs to be changed to something like 
```TXT
This patch uses `memcpy()` to copy all the content from another BitArray instead of applying multiple bit operations.
```

> But then this in guard is broken here.

Why? `len` is the number of bits in the `BitArray`